### DISCUSSION TITLE
Is it possible to background `quarto preview` and get the pid (to kill it) in a file?

### DISCUSSION THREAD

**Discussion by rkrug** [quarto-dev/quarto-cli#457] [Q&A]
Hi

os: macOS

I am trying to start quarto in the background in one script and save the pid in a file, while in a second script to read that pid and kill it. 

This is more difficult than I expected, as the quarto process (of which I can get the pid) launches a quarto.js process (of which I don't manage to get the pid), and even if I ill the quarto process, the quarto.js process continues.

My question:

- is it possible to save the backgrounded quarto (and quarto.js) process pid, so that I can kill them when not needed anymore?
- would it be possible to include a `--background-start` and a `--background-kill` option for tat purpose, to not having to resort to bash scripting?

Thanks,

Rainer

**Comment by jjallaire** [comment]
We have had no problem killing the quarto process by sending it a negative signal value (to indicate that you want to kill the whole process group). e.g. `kill -9 <pid>` Have you tried that?
